src/GoTrueApi.ts

```diff
@@ -32,6 +32,46 @@ export default class GoTrueApi {
     this.cookieOptions = { ...COOKIE_OPTIONS, ...cookieOptions }
   }
 
+  /**
+   * Creates a new user.
+   *
+   * This function should only be called on a server. Never expose your `service_role` key in the browser.
+   *
+   * @param attributes The data you want to create the user with.
+   * @param jwt A valid JWT. Must be a full-access API key (e.g. service_role key).
+   */
+  async createUser(
+    attributes: UserAttributes
+  ): Promise<{ data: null; error: ApiError } | { data: User; error: null }> {
+    try {
+      const data: any = await post(`${this.url}/admin/users`, attributes, {
+        headers: this.headers,
+      })
+      return { data, error: null }
+    } catch (e) {
+      return { data: null, error: e as ApiError }
+    }
+  }
+
+  /**
+   * Get a list of users.
+   *
+   * This function should only be called on a server. Never expose your `service_role` key in the browser.
+   *
+   * @param attributes The data you want to create the user with.
+   * @param jwt A valid JWT. Must be a full-access API key (e.g. service_role key).
+   */
+  async listUsers(): Promise<{ data: null; error: ApiError } | { data: User[]; error: null }> {
```

Comment:
The response data is actually of the format `{ aud: "authenticated", users: User[] }`.

Ref: https://github.com/supabase/gotrue/blob/401d4cf375e941f3db0fff84319e56a2bdec8d26/api/admin.go#L80-L83.